spec/unit/transaction_spec.rb

```diff
@@ -613,6 +625,11 @@ def transaction_with_resource(resource)
         it "should not rescue prefetch executions" do
           expect { transaction.prefetch_if_necessary(resource) }.to raise_error(Puppet::Error)
         end
+
+        it "should log the exception during prefetch" do
+          Puppet.expects(:log_exception)
+          expect { transaction.prefetch_if_necessary(resource) }.to raise_error(Puppet::Error)
+        end
```

Comment:
Could you update both expectations to include the message, so the test doesn't silently pass if an unexpected `Puppet::Error` is raised (unrelated to this issue)?